pkg/nsx/services/vpc/store.go

```diff
@@ -85,10 +89,18 @@ func (vs *VPCStore) GetVPCsByNamespace(ns string) []*model.Vpc {
 	for _, vpc := range vpcs {
 		mvpc := vpc.(*model.Vpc)
 		tags := mvpc.Tags
+		nameMatch := false
+		idMatch := false
 		for _, tag := range tags {
-			if *tag.Scope == common.TagScopeNamespace && *tag.Tag == ns {
-				ret = append(ret, mvpc)
+			if *tag.Scope == common.TagScopeNamespace && *tag.Tag == ns.Name {
```

Comment:
Why not update the `VPCStore` to configure with NS UID as index in `InitializeVPC`, then here we can directly use UUID as index to search the VPCs.

Reply:
Yes, if that, we should add a new indexer function in VPC store, currently, there is VPC's ID as key in the VPC store.  We need to consider there are multiple NS UID for multiple VPCs, i.e, shared VPCs, also in the future, we might need to support one NS UID for multiple VPCs.

Reply:
Implement the VPC index in https://github.com/vmware-tanzu/nsx-operator/pull/825